api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -477,10 +477,10 @@ message CertificateValidationContext {
   // certificate chain will be subject to validation by :ref:`CRL <envoy_v3_api_field_extensions.transport_sockets.tls.v3.CertificateValidationContext.crl>`.
   bool only_verify_leaf_cert_crl = 14;
 
-  // Config for the max number of intermediate certificates in chain that are parsed during verification.
-  // This does not include the leaf certificate. If configured, and the certificate chain is longer than allowed, the certificates
-  // above the limit are ignored, and certificate validation will fail. The default limit is 100,
-  // though this can be system-dependent.
-  // https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_verify_depth.html
+  // Defines the max number of intermediate certificates in the chain that are parsed during verification.
+  // This does not include the leaf certificate. If a trusted issuer appears in the chain, but in a depth larger than configured, the certificate validation will fail.
```

Comment:
instead of saying `This does not include the leaf certificate.`, we can say that `the max_verify_depth defines the limit on the number of certificates between the end-entity and trust-anchor certificates. the trust-anchor was specified at 'trust_ca' https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-certificatevalidationcontext`